source/common/http/conn_pool_grid.cc

```diff
@@ -334,5 +333,50 @@ void ConnectivityGrid::onDrainReceived() {
   }
 }
 
+bool ConnectivityGrid::shouldAttemptHttp3() {
+  if (http3_status_tracker_.isHttp3Broken()) {
+    ENVOY_LOG(trace, "HTTP/3 is broken to host '{}', skipping.", host_->hostname());
+    return false;
+  }
+  if (!alternate_protocols_.has_value()) {
+    // TODO(RyanTheOptimist): Make the alternate protocols cache required.
```

Comment:
nit: how about this? `// TODO(RyanTheOptimist): Do not attempt HTTP/3 when the alternate protocols cache is missing.`

Reply:
Well, I could do that but that's not the goal of this TODO :) The goal of the TODO is that the Grid is always constructed with an AlternateProtocolsCache. I can't easily do that as part of this PR (I'll be doing in 2 PRs down the road) because in order to do this, we need to construct per-config AlternateProtocolCaches which requires a bunch of plumbing. I have that queued up, but it's a lot of unrelated work.

Reply:
Ah I see, in that case how about `// TODO(RyanTheOptimist): Make the alternate_protocols_ member non-optional.` Additionally, maybe move the TODO to the definition of that member variable?